x-pack/plugins/secops/public/components/page/network/kpi_network/index.tsx

```diff
@@ -22,34 +23,44 @@ import * as i18n from './translations';
 
 interface KpiNetworkProps {
   data: KpiNetworkData;
+  loading: boolean;
 }
 
-const kpiNetworkCards = (data: KpiNetworkData) => [
+const loadMatrix = (loading: boolean, data: KpiNetworkData, property: string) => {
+  const matrix: number | null | undefined = get(property, data);
+
+  if (typeof matrix !== 'undefined' && matrix !== null) {
+    return numeral(matrix).format('0,0');
+  } else {
+    if (loading) {
+      return <EuiLoadingSpinner size="m" />;
+    } else {
+      getEmptyTagValue();
+    }
+  }
+};
+
+const kpiNetworkCards = (loading: boolean, data: KpiNetworkData) => [
   {
-    title:
-      has('networkEvents', data) && data.networkEvents !== null
-        ? numeral(data.networkEvents).format('0,0')
-        : getEmptyTagValue(),
+    title: loadMatrix(loading, data, 'networkEvents'),
```

Comment:
++ on the refactoring here :) We'll have a few more KPI components it looks like, so you could go even further and abstract the fieldName + i18n key into a `fieldTitleMapping` and loop over them to generate these cards and make things even more generic, but that can happen organically over time as it makes sense. Either way, nice cleanup + additions w/ `loading` as well!